### ISSUE TITLE
`bugprone-nondeterministic-pointer-iteration-order` reported with custom comparator

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#145665]
```cpp
#include <algorithm>
#include <vector>

struct S
{
    int i() const;
};

void f()
{
    std::vector<const S*> v;
    std::sort(v.begin(), v.end(), [](const S* s1, const S* s2){
        return s1->i() > s2->i();
    });
}
```

```
<source>:12:5: warning: sorting pointers is nondeterministic [bugprone-nondeterministic-pointer-iteration-order]
   12 |     std::sort(v.begin(), v.end(), [](const S* s1, const S* s2){
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   13 |         return s1->i() > s2->i();
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~
   14 |     });
```
https://godbolt.org/z/xjYe63rY6

This is only reported with libc++.

The warning should not be reported if a custom compare function is provided.

**Comment by PiotrZSL** [comment]
This is because check only verify that "v" got field with pointer to pointer, for me this check need to be re-implemented more properly.
